backend/tournesol/views/polls_reco_random.py

```diff
@@ -0,0 +1,88 @@
+    extend_schema_view,
+)
+
+from tournesol.models import Entity
+from tournesol.serializers.poll import (
+    RecommendationBaseSerializer,
+    RecommendationsRandomFilterSerializer,
+)
+from tournesol.utils.cache import cache_page_no_i18n
+from tournesol.views import PollRecommendationsBaseAPIView
+
+
+class RandomRecommendationBaseAPIView(PollRecommendationsBaseAPIView):
+    def get_queryset(self):
+        """
+        Return a queryset of random recommended entities.
+
+        This queryset is designed to be more performant than the queryset
+        of the regular recommendations view. For this reason, it doesn't allow
+        to:
+            - filter entities by text
+            - filter entities by weighted criteria score
+            - or anything involving a SQL JOIN on EntityCriteriaScore
+        """
+        poll = self.poll_from_url
+        queryset = Entity.objects.all()
+        queryset, _ = self.filter_by_parameters(self.request, queryset, poll)
+        queryset = queryset.with_prefetched_poll_ratings(poll_name=poll.name)
+        queryset = queryset.filter_safe_for_poll(poll)
+        queryset = queryset.order_by("?")
+        return queryset
+
+
+@extend_schema_view(
+    get=extend_schema(
+        parameters=[
+            RecommendationsRandomFilterSerializer,
+            OpenApiParameter(
+                "metadata",
+                OpenApiTypes.OBJECT,
+                style="deepObject",
+                description="Filter by one or more metadata.",
+                examples=[
+                    OpenApiExample(
+                        name="No metadata filter",
+                    ),
+                    OpenApiExample(
+                        name="Videos - some examples",
+                        value={"language": ["en", "pt"], "uploader": "Kurzgesagt – In a Nutshell"},
+                    ),
+                    OpenApiExample(
+                        name="Videos - videos of 8 minutes or less (480 sec)",
+                        value={"duration:lte:int": "480"},
+                    ),
+                    OpenApiExample(
+                        name="Candidates - some examples",
+                        value={
+                            "name": "A candidate full name",
+                            "youtube_channel_id": "channel ID",
+                        },
+                    ),
+                ],
+            ),
+        ],
+    )
+)
+class RandomRecommendationList(RandomRecommendationBaseAPIView):
+    """
+    Return a random list of recommended entities.
+    """
+
+    permission_classes = []
+    serializer_class = RecommendationBaseSerializer
+    poll_parameter = "name"
+
+    @method_decorator(cache_page_no_i18n(60 * 10))
+    def get(self, request, *args, **kwargs):
+        response = self.list(self, request, *args, **kwargs)
+        if request.query_params.get("exclude_compared_entities") == "true":
+            patch_vary_headers(response, ["Authorization"])
```

Comment:
Doesn't this line means that the cache is separate between the users. I.e. if 1000 users request that route once. The cache will contain 1000 entries and never successfully hit?

Reply:
Yes, this is something I still need to fix.

Reply:
Fixed by https://github.com/tournesol-app/tournesol/pull/1876/commits/cf9f06202b38b6f2af4468f11a8ae8ad9422b039